### ISSUE TITLE
Will google block this app after January 4, 2021?

### ISSUE THREAD

**Issue by marcocarnazzo** [googlesamples/google-signin-unity#168]
Google send to us a mail saying: "Starting January 4, 2021, we will block all sign-ins to Google accounts from embedded browser frameworks"
Our affected app use this Google Signin library: it will work after January 4, 2021?

**Comment by derwaldgeist** [comment]
Is this using an embedded browser framework? I thought it was native.

**Comment by adiohaion** [comment]
I would be happy to get an answer to that also if someone knows.

**Comment by derwaldgeist** [comment]
Absolutely! It's a shame that this package seems so abandoned by Google. Why do they release such stuff if they don't care for maintaining it? We just decided to not go live with it because of this uncertainty and rather wait until January to be sure.

**Comment by michal-lorenc** [comment]
I don't think it will affect this plugin.
From what i understood it turns out that you will not be able to login to google inside web view.
But when you sign in with this plugin it doesn't open a web browser, just a single login pop up like in every even pure java app.
Im not sure, but i think we are safe guys 👍

**Comment by zeegfa** [comment]
> 
> 
> I don't think it will affect this plugin.
> From what i understood it turns out that you will not be able to login to google inside web view.
> But when you sign in with this plugin it doesn't open a web browser, just a single login pop up like in every even pure java app.
> Im not sure, but i think we are safe guys 👍

What is doubtful is that it is involved in the Apple web browser system and therefore creates a problem during the build cuz using UI web view which is deprecated from iOS... can be solved with #126 , but the question is whether there is anything equivalent in Android Or not?

**Comment by derwaldgeist** [comment]
> i think we are safe guys

Awesome, sounds good! I just wasn't sure because the popup kinda looks like a web page. But yes, the native libraries are included as well, so I hope you're right.

**Comment by derwaldgeist** [comment]
> whether there is anything equivalent in Android Or not

Not to my knowledge. Apple deprecated their old UIWebView i favour of WkWebView, so that's mainly an Apple thing. And I can confirm that Apple accepts apps that are based upon the fork referenced in #126. We're already using this version in our app marblear.com, although we've disabled Google Sign-in in the UI so far because of the doubts we had about this issue here. If there would be a problem in the build because of references to UIWebView, they would have noticed.

**Comment by marcocarnazzo** [comment]
> From what i understood it turns out that you will not be able to login to google inside web view.
> But when you sign in with this plugin it doesn't open a web browser, just a single login pop up like in every even pure java app.

Thank you.
I also think so: i created this issue just to receive a confirm.
Did you received like me a mail alert from Google about your Unity project?

PS: Maybe we are safe but this projects seems abandoned (i e. no official reply in this thread) and it's a little scary :-( .

**Comment by michal-lorenc** [comment]
> Did you received like me a mail alert from Google about your Unity project?

I did not receive such an email, but I did not publish my app yet.

Maybe it's a problem related to iOS?
Is your app published for android & iOS?


> PS: Maybe we are safe but this projects seems abandoned (i e. no official reply in this thread) and it's a little scary :-( .

Yeah, it is abandoned. This package was lastly updated 2 or 3 years ago. From what i see there is no real support for this package anymore. It uses really old google libraries.


I was tired of no support for this plugin and of some bugs I didn't know how to fix, so I made my own plugin for unity in java and it works perfectly, but probably supports only android.

**Comment by adiohaion** [comment]
Though this plugin works well for now, could be better to migrate to https://firebase.google.com/docs/auth/unity/google-signin

**Comment by michal-lorenc** [comment]
@adiohaion Fire base auth requires google sign in plugin so it does not fix anything.

**Comment by adiohaion** [comment]
My bad put the wrong link - https://github.com/playgameservices/play-games-plugin-for-unity - meant to just migrate to google play services (even if your app is not a game probably worth it). Google probably prefer that when using unity, guess thats why this plugin is not maintained.

**Comment by IdoMarkovitz** [comment]
> My bad put the wrong link - https://github.com/playgameservices/play-games-plugin-for-unity - meant to just migrate to google play services (even if your app is not a game probably worth it). Google probably prefer that when using unity, guess thats why this plugin is not maintained.

It's no longer supported for iOS (deprecated) so that doesn't solve it either.

**Comment by michal-lorenc** [comment]
Idk my brothers, i wrote my own google sign-in plugin for unity but it supports only android as I don't have iPhone.

**Comment by riak157** [comment]
Was that difficult? Looking into possibly doing this myself.

**Comment by michal-lorenc** [comment]
@riak157 not really. Once you understand it, it's easy. The most important part you need to know is that, if the plugin you made use another plugin, then you need to include it also in unity project, i spent days trying to figure out why it did not work.

Here are some links that might help you:
1. How to make android plugin in java for unity http://markcastle.com/steps-to-create-a-native-android-plugin-for-unity-in-java-using-android-studio-part-1-of-2/
2. Making google sign in with pure java https://developers.google.com/identity/sign-in/android/start-integrating

**Comment by riak157** [comment]
Alright thanks! I appreciate it. How come you made your own plugin rather than using this one?

**Comment by michal-lorenc** [comment]
Last update was 3 years ago. Some things changed by this time. Unfortunetly this plugin is abandoned and nobody really cares about updating it anymore. 

I had to make integration with Google Fit API. I believe that would be impossible without access to 'google sign in' source code.

Sometimes it's easier to make things from scratch than using outdated & unsupported plugin.

(the amount of opened and unsolved issues might give you a good view why it's not worth your time)

**Comment by derwaldgeist** [comment]
Can anybody confirm that this plugin is still accepted by the Play Store? We have successfully implemented it, but wanted to wait until we know if it was blocked before we release it in our app.

**Comment by popisdead** [comment]
lol! hey @derwaldgeist, how are you? Funny to find it here :) - better to login probably through html5 browser, this plugin seems to cause too much problem. Too bad people don't remember passwords.

**Comment by lucashaley** [comment]
> I had to make integration with Google Fit API. I believe that would be impossible without access to 'google sign in' source code.
> 

@Xazerekt did you get Google Fitness working from Unity?  I'd love to know if it's possible…

**Comment by michal-lorenc** [comment]
@lucashaley Yes it is. Unity, C# + Java

**Comment by jeisoncp** [comment]
> Can anybody confirm that this plugin is still accepted by the Play Store? We have successfully implemented it, but wanted to wait until we know if it was blocked before we release it in our app.

Yes @derwaldgeist, is still been accepted on Play Store. And still works for Android, at least.
